packages/jest-mock/src/__tests__/jest_mock.test.js

```diff
@@ -180,9 +183,140 @@ describe('moduleMocker', () => {
+      expect(instanceFooMock.toString.mock).toBeDefined();
+    });
+
+    it('mocks ES2015 non-enumerable static properties and methods', () => {
+      class ClassFoo {
+        static foo() {}
+      }
+      ClassFoo.fooProp = () => {};
+
+      class ClassBar extends ClassFoo {}
+
+      const ClassBarMock = moduleMocker.generateFromMetadata(
+        moduleMocker.getMetadata(ClassBar),
+      );
+
+      expect(typeof ClassBarMock.foo).toBe('function');
+      expect(typeof ClassBarMock.fooProp).toBe('function');
+      expect(ClassBarMock.foo.mock).toBeDefined();
+      expect(ClassBarMock.fooProp.mock).toBeDefined();
+    });
+
+    it('mocks methods in all the prototype chain (null prototype)', () => {
+      const Foo = Object.assign(Object.create(null), {foo() {}});
+      const Bar = Object.assign(Object.create(Foo), {bar() {}});
+
+      const BarMock = moduleMocker.generateFromMetadata(
+        moduleMocker.getMetadata(Bar),
+      );
+      expect(typeof BarMock.foo).toBe('function');
+      expect(typeof BarMock.bar).toBe('function');
+    });
+
+    it('does not mock methods from Object.prototype', () => {
+      const Foo = {foo() {}};
+      const Bar = Object.assign(Object.create(Foo), {bar() {}});
+
+      const BarMock = moduleMocker.generateFromMetadata(
+        moduleMocker.getMetadata(Bar),
+      );
+
+      expect(BarMock).toBeInstanceOf(mockGlobals.Object);
+      expect(
+        Object.prototype.hasOwnProperty.call(BarMock, 'hasOwnProperty'),
+      ).toBe(false);
+      expect(BarMock.hasOwnProperty).toBe(
+        mockGlobals.Object.prototype.hasOwnProperty,
+      );
+    });
+
+    it('does not mock methods from Object.prototype (in mock context)', () => {
+      const Bar = vm.runInContext(
+        `
+        const Foo = { foo() {} };
+        const Bar = Object.assign(Object.create(Foo), { bar() {} });
+        Bar;
+      `,
+        mockContext,
+      );
+
+      const BarMock = moduleMocker.generateFromMetadata(
+        moduleMocker.getMetadata(Bar),
+      );
+
+      expect(BarMock).toBeInstanceOf(mockGlobals.Object);
+      expect(
+        Object.prototype.hasOwnProperty.call(BarMock, 'hasOwnProperty'),
+      ).toBe(false);
+      expect(BarMock.hasOwnProperty).toBe(
+        mockGlobals.Object.prototype.hasOwnProperty,
+      );
+    });
+
+    it('does not mock methods from Function.prototype', () => {
+      class Foo {}
+      class Bar extends Foo {}
+
+      const BarMock = moduleMocker.generateFromMetadata(
+        moduleMocker.getMetadata(Bar),
+      );
+
+      expect(BarMock).toBeInstanceOf(mockGlobals.Function);
+      expect(Object.prototype.hasOwnProperty.call(BarMock, 'bind')).toBe(false);
+      expect(BarMock.bind).toBe(mockGlobals.Function.prototype.bind);
+    });
+
+    it('does not mock methods from Function.prototype (in mock context)', () => {
+      const Bar = vm.runInContext(
+        `
+        class Foo {}
+        class Bar extends Foo {}
+        Bar;
+      `,
```

Comment:
There's a weird alignment going on here... Do instead:

```javascript
      const Bar = vm.runInContext(
        `
          class Foo {}
          class Bar extends Foo {}
          Bar;
        `,
        mockContext,
      );
```

Reply:
That's all prettier 🤷🏻‍♂️

Reply:
you can add a space inside the literal to align the backticks, prettier won't strip them

Reply:
Correct; once you're in backtick mode, prettier is not changing the contents.